functions/__fzf_search_current_dir.fish

```diff
@@ -8,10 +8,18 @@ function __fzf_search_current_dir --description "Search the current directory us
     )
 
     if test $status -eq 0
-        # If this function was triggered with an empty commandline and the only thing selected is a directory, then
-        # prepend ./ to the dir path. Because fish will attempt to cd implicitly if a directory name starting with a dot
-        # is provided, this allows the user to hit Enter one more time to quickly cd into the selected directory.
-        if test (count (commandline --tokenize)) = 0 && test (count $file_paths_selected) = 1 && test -d $file_paths_selected
+        # If this function was triggered when the user is inputing the first token and only one path is selected,
+        # then prepend ./ to the selected path.
+        # If the path is an executable, the user can hit Enter one more time to execute it.
+        # If the path is a directory, the user can hit Enter one more time to quickly cd into it, because fish will
+        # attempt to cd implicitly if a directory name starts with a dot.
+        set first_token_length (string length (commandline --tokenize)[1])
+
+        if test -z "$first_token_length"
+            set first_token_length 0
+        end
+
+        if test (string length (commandline)) = $first_token_length && test (count $file_paths_selected) = 1
```

Comment:
Can you reverse the order of the if statement for performance reasons?

Reply:
Also, can't you just `test (count (commandline --tokenize) = 1)` to tell if the user only inputted one token?

Reply:
> Can you reverse the order of the if statement for performance reasons?

As you wish, but two builtins vs one builtin doesn't make any difference.

Reply:
> just `test (count (commandline --tokenize) = 1)` to tell if the user only inputted one token

- This doesn't allow it being triggered with an empty command line
- `<command> |` (`|` indicates cursor position) is considered one token by `commandline --tokenize`, which is not desirable.

Reply:
Oh good point, forgot commandline was a builtin too.

I still find the logic convoluted though. If all we want to know is if the user is still typing the first token, can't we just compare (commandline) to (commandline --current-token)?

Reply:
Or even `test (commandline --tokenize) = (commandline --current-token)`

Reply:
> `test (commandline --tokenize) = (commandline --current-token)`

Brilliant! We can even drop `--tokenize`.

Reply:
Actually, we should keep tokenize for situations where the user accidentally included a whitespace in the front or back for whatever reason but still has the cursor on the current token. I'll add it in and test.

Reply:
Oh darn, now it doesn't work if the input is empty. Hmm...

Reply:
Oh it's because if the commandline is empty, then (commandline --tokenize) is empty and test doesn't work
```
test: Missing argument at index 2

~/.config/fish/functions/__fzf_search_current_dir.fish (line 29):
        if test (count $file_paths_selected) = 1 && test ""(commandline --tokenize) = ""(commandline --current-token)
                                                    ^
in function '__fzf_search_current_dir'
```